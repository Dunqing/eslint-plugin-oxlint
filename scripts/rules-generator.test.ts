import { vi, test, suite, vitest, beforeEach, afterEach, expect } from 'vitest';
import { RulesGenerator, RulesGrouping } from './rules-generator.js';
import type { Rule } from './traverse-rules.js';

suite('RulesGenerator', () => {
  beforeEach(() => {
    vi.mock('./oxlint-version.ts', () => ({
      getLatestVersionFromClonedRepo: vitest.fn().mockReturnValue('1.0.0'),
    }));
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  test('RulesGenerator generates rules correctly', async () => {
    const successResultArray: Rule[] = [
      {
        category: 'style',
        scope: 'eslint',
        value: 'rulename-with-mod',
      },
      {
        category: 'correctness',
        scope: 'typescript',
        value: '@typescript-eslint/rulename-without-mod',
      },
    ];

    // Create an instance of RulesGenerator
    const generator = new RulesGenerator(
      successResultArray,
      RulesGrouping.SCOPE
    );

    // Call the generateRules method
    expect(await generator.generateRulesCode()).toMatchSnapshot('byScope');

    generator.setRulesGrouping(RulesGrouping.CATEGORY);
    // Call the generateRules method
    expect(await generator.generateRulesCode()).toMatchSnapshot('byCategory');
  });
});
