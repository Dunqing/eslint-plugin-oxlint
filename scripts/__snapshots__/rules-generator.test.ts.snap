// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RulesGenerator > RulesGenerator generates rules correctly > byCategory 1`] = `
"// These rules are automatically generated by scripts/generate-rules.ts

const styleRules = {
  'rulename-with-mod': "off"
}

const correctnessRules = {
  '@typescript-eslint/rulename-without-mod': "off"
}

export {
  styleRules,
  correctnessRules
}"
`;

exports[`RulesGenerator > RulesGenerator generates rules correctly > byScope 1`] = `
"// These rules are automatically generated by scripts/generate-rules.ts

const eslintRules = {
  'rulename-with-mod': "off"
}

const typescriptRules = {
  '@typescript-eslint/rulename-without-mod': "off"
}

export {
  eslintRules,
  typescriptRules
}"
`;
